app/src/org/commcare/android/database/connect/models/PushNotificationRecord.kt

```
@@ -65,12 +68,45 @@ class PushNotificationRecord : Persisted(), Serializable {
         const val META_ACTION = "action"
         const val META_TITLE = "title"
         const val META_BODY = "body"
-        const val META_DATE_TIME = "date_time"
+        const val META_TIME_STAMP = "timestamp"
         const val META_CONFIRMATION_STATUS = "confirmation_status"
         const val META_OPPORTUNITY_ID = "opportunity_id"
         const val META_MESSAGE_ID = "message_id"
         const val META_CHANNEL = "channel"
         const val META_PAYMENT_ID = "payment_id"
         const val META_READ_STATUS = "read_status"
+
+        fun fromJsonArray(jsonArray: JSONArray): List<PushNotificationRecord> {
+            val records = mutableListOf<PushNotificationRecord>()
+
+            for (i in 0 until jsonArray.length()) {
+                try {
+                    val obj = jsonArray.getJSONObject(i)
+                    val record = PushNotificationRecord().apply {
+                        notificationId = obj.optString(META_NOTIFICATION_ID,"")
+                        notificationType = obj.optString(META_NOTIFICATION_TYPE,"")
+                        title = obj.optString(META_TITLE,"")
+                        body = obj.optString(META_BODY,"")
+                        confirmationStatus = obj.optString(META_CONFIRMATION_STATUS,"")
+                        paymentId = obj.optString(META_PAYMENT_ID,"")
+                        readStatus = obj.optBoolean(META_READ_STATUS,false)
+                        timeStamp = obj.optString(META_TIME_STAMP,"")
+
+                        val dataObj = obj.optJSONObject("data")
+                        dataObj?.let {
+                            connectMessageId = it.optString(META_MESSAGE_ID,"")
+                            channel = it.optString(META_CHANNEL,"")
+                            action = it.optString(META_ACTION,"")
```

@jaypanchal-13 action is mandatory field and should raise exception if not available